### PR TITLE
Upgrade GCP secretmanager from v1beta1 to v1

### PR DESCRIPTION
**Pull request by abursavich**
I have a transitive dependency on vals, which is causing problems upgrading my other dependencies because Google has removed the GCP secretmanager v1beta1 APIs from recent modules. This change migrates from v1beta1 to v1.

### CODE REVIEW

**Comment by abursavich** [comment]
Rebased to resolve conflicts.

**Comment by abursavich** [comment]
PTAL @mumoshu

File: pkg/providers/gcpsecrets/gcpsecrets.go [hunk 1]
```diff
@@ -33,9 +31,10 @@ func New(cfg api.StaticConfig) *provider {
 
 	version := cfg.String("version")
 	if version == "" {
-		p.version = "latest"
-	} else {
-		p.version = version
+		version = "latest"
+	}
+	return &provider{
+		version: version,
```
Comment (yxxhero) [inline pkg/providers/gcpsecrets/gcpsecrets.go:37]: @mumoshu return? how about the below code?

Comment (mumoshu) [reply pkg/providers/gcpsecrets/gcpsecrets.go:37]: This seems terribly wrong! Let's revert this part so that the rest of the New func works !

Comment (abursavich) [reply pkg/providers/gcpsecrets/gcpsecrets.go:37]: It looks like several different changes got merged in this file on the same day. The original PR looks clean to me
https://github.com/abursavich/vals/commit/67f05de6f50052f02eb280271783013a4ce6a2e4. Sorry for the trouble.

Comment (abursavich) [reply pkg/providers/gcpsecrets/gcpsecrets.go:37]: Here's a fix: https://github.com/helmfile/vals/pull/128

**Pull request merged**